src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs

```diff
@@ -26,26 +26,32 @@ public interface IJoinFederationRequestService
     public sealed class JoinFederationRequestService : IJoinFederationRequestService
     {
         private readonly ICounterChainSettings counterChainSettings;
+        private readonly IFederationManager federationManager;
         private readonly IFullNode fullNode;
         private readonly IHttpClientFactory httpClientFactory;
-        private readonly ILoggerFactory loggerFactory;
+        private readonly ILogger logger;
         private readonly PoANetwork network;
         private readonly NodeSettings nodeSettings;
         private readonly VotingManager votingManager;
 
-        public JoinFederationRequestService(ICounterChainSettings counterChainSettings, IFullNode fullNode, IHttpClientFactory httpClientFactory, ILoggerFactory loggerFactory, Network network, NodeSettings nodeSettings, VotingManager votingManager)
+        public JoinFederationRequestService(ICounterChainSettings counterChainSettings, IFederationManager federationManager, IFullNode fullNode, IHttpClientFactory httpClientFactory, Network network, NodeSettings nodeSettings, VotingManager votingManager)
         {
             this.counterChainSettings = counterChainSettings;
+            this.federationManager = federationManager;
             this.fullNode = fullNode;
             this.httpClientFactory = httpClientFactory;
-            this.loggerFactory = loggerFactory;
+            this.logger = LogManager.GetCurrentClassLogger();
             this.network = network as PoANetwork;
             this.nodeSettings = nodeSettings;
             this.votingManager = votingManager;
         }
 
         public async Task<PubKey> JoinFederationAsync(JoinFederationRequestModel request, CancellationToken cancellationToken)
         {
+            // First ensure that this collateral address isnt already present in the federation.
```

Comment:
A parallel request could already be in progress and this test would not catch that?

Reply:
We could check the pending polls rather?

Reply:
I think for now this is sufficient. Once we bring in your polls repository code we can check pending polls as it is possible that pending duplicate polls do exist.

Reply:
We should be able to easily prevent multiple pending polls involving the same collateral address.